src/hooks/auditoriumSections.ts

```diff
@@ -0,0 +1,134 @@
+import { User } from "types/User";
+
+import { setGridData } from "api/profile";
+
+import {
+  currentAuditoriumSectionsSelector,
+  currentAuditoriumSectionsByIdSelector,
+} from "utils/selectors";
+import { WithId } from "utils/id";
+import { getPositionHash } from "utils/auditorium";
+
+import { useSelector } from "./useSelector";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useRecentVenueUsers } from "./users";
+import { useUser } from "./useUser";
+import { useCallback, useMemo } from "react";
+
+const useConnectAuditoriumSections = (venueId?: string) => {
+  useFirestoreConnect(() => {
+    if (!venueId) return [];
+
+    return [
+      {
+        collection: "venues",
+        doc: venueId,
+        subcollections: [{ collection: "sections" }],
+        storeAs: "currentAuditoriumSections",
+      },
+    ];
+  });
+};
+
+export interface UseAuditoriumSectionProps {
+  sectionId: string;
+  venueId?: string;
+}
+
+export const useAuditoriumSection = ({
+  venueId,
+  sectionId,
+}: UseAuditoriumSectionProps) => {
+  useConnectAuditoriumSections(venueId);
+
+  const { userWithId } = useUser();
+  const userId = userWithId?.id;
+
+  const sections = useSelector(currentAuditoriumSectionsByIdSelector);
+  const section = sections?.[sectionId];
+
+  const seatedUsers = useSectionSeatedUsers(venueId, sectionId);
+
+  const seatedUsersByHash = useMemo(
+    () =>
+      seatedUsers.reduce<Record<string, WithId<User> | undefined>>(
+        (acc, user) => {
+          if (!venueId) return acc;
+
+          const takenRow = user.data?.[venueId].row;
+          const takenColumn = user.data?.[venueId].column;
+
+          if (takenRow === undefined || takenColumn === undefined) return acc;
+
+          const positionHash = getPositionHash({
+            row: takenRow,
+            column: takenColumn,
+          });
+
+          return { ...acc, [positionHash]: user };
+        },
+        {}
+      ),
+    [seatedUsers, venueId]
+  );
+
+  const getUserBySeat = useCallback(
+    ({ row, column }: { row: number; column: number }) =>
+      seatedUsersByHash && seatedUsersByHash[getPositionHash({ row, column })],
+    [seatedUsersByHash]
+  );
+
+  const takeSeat = useCallback(
+    ({ row, column }: { row: number; column: number }) => {
```

Comment:
This is another place where my suggested GridPosition utility type could be used in place of directly spelling it out in the function params.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1251/commits/75ae7353c6a7e5d30d83b9f7309d8c755ee318c1